### ISSUE TITLE
Router links not resilient to controller crash

### ISSUE THREAD

**Issue by arunsworld** [openziti/ziti#1471]
After a controller crash (#1460) and recovery; the router link does not recover.

The router reconnects and it can be seen as ONLINE in `ziti fabric list routers`.

However, the router establishes no links.

Router logs:
```
[55866.180]    INFO ziti/router/handler_edge_ctrl.(*apiSessionAddedHandler).applySync: finished sychronizing api sessions [count: 3, syncId: clo6y9p460004s4p767smr4yq, duration: 58.666µs]
[55866.204]    INFO ziti/router/handler_link.(*bindHandler).BindChannel: {linkId=[2DWoLM0vGdDz9vR2K4xDgu] routerId=[lT--ORmNUG] routerVersion=[v0.0.0]} link destination support heartbeats
[55866.204]    INFO ziti/router.(*xlinkAccepter).Accept: accepted new link [l/2DWoLM0vGdDz9vR2K4xDgu]
[55866.204] WARNING ziti/router/link.(*linkRegistryImpl).evaluateLinkState.func1: {key=[default->tls:lT--ORmNUG->default]} link already present, but link status still pending
[55866.263]    INFO ziti/router/handler_ctrl.(*faultHandler).handleFault [ch{ctrl}->u{reconnecting}->i{QMOY}]: {linkId=[2DWoLM0vGdDz9vR2K4xDgu]} link fault reported, closing
---
[56168.889]    INFO ziti/router/handler_link.(*closeHandler).HandleClose [ch{l/2DWoLM0vGdDz9vR2K4xDgu}->u{classic}->i{zDq1}]: {routerId=[lT--ORmNUG] linkId=[2DWoLM0vGdDz9vR2K4xDgu]} link closed
---
[56168.893] WARNING ziti/router/link.(*linkRegistryImpl).evaluateLinkState.func1: {key=[default->tls:lT--ORmNUG->default]} link already present, but link status still pending
[56169.752] WARNING ziti/router/link.(*linkRegistryImpl).evaluateLinkState.func1: {key=[default->tls:lT--ORmNUG->default]} link already present, but link status still pending
[56169.754] WARNING ziti/router/link.(*linkRegistryImpl).evaluateLinkState.func1: {key=[default->tls:lT--ORmNUG->default]} link already present, but link status still pending
```

**Comment by arunsworld** [comment]
Update: discovered that disabling and then re-enabling the router recovers the link:

```
ziti fabric update router 5Wd7iovsP --disabled
ziti fabric update router 5Wd7iovsP --disabled=false
```

I've also been looking to see if there are ways to debug misbehaving routers remotely. I've discovered the fabric inspect commands. But if there are specific ones to help debug when expected links go missing, would appreciate it.

**Comment by arunsworld** [comment]
Spoke too soon; even though this appears to restore the link (shows up in ```fabric list links```), trying to use it fails (cannot create circuit) and then disappears from the listing.

```fabric inspect links``` shows the link as ```status: dialing``` and appears to be in that state permanently.

Router logs:
```
[121547.405]    INFO transport/v2/tcp.acceptLoop [tcp/tcp:127.0.0.1:1125]: {addr=[127.0.0.1:54281]} accepted connection
[121547.565]   ERROR ziti/router/handler_ctrl.(*routeHandler).fail |establishPath|: {error=[invalid link destination 1RZVffuKVFQO3IQdxPskMf] context=[ch{ctrl}->u{reconnecting}->i{eZ8D}] attempt=[0] circuitId=[rRraBBRZa] serviceId=[7E9m60gdfj3qwScy5G0663] attemptNumber=[1]} failure while handling route update
[121547.566]    INFO ziti/router/handler_ctrl.(*faultHandler).handleFault [ch{ctrl}->u{reconnecting}->i{eZ8D}]: {linkId=[1RZVffuKVFQO3IQdxPskMf]} link fault reported, link already closed or unknown
[121547.600]   ERROR ziti/router/xgress.GetCircuit: {error=[<nil>]} failure creating circuit
[121547.600]   ERROR ziti/router/xgress_proxy.(*listener).handleConnect [tcp:127.0.0.1:54281 <- {tcp}]: error creating circuit (can't route from 5Wd7iovsP -> .TB3go-dA)
```

**Comment by plorenz** [comment]
Digging into this today

**Comment by plorenz** [comment]
I did some additional chaos testing and was able to reproduce this. I added fixes to the `router-hello-panic-debug` branch, to make tesing easier, but I'll be working on getting those merged up independently.

**Comment by plorenz** [comment]
FYI: Fix released on 0.31.0
